python/tests/transformers/tf_tensor_test.py

```diff
@@ -0,0 +1,323 @@
+#
+from __future__ import absolute_import, division, print_function
+
+import os
+import shutil
+import tempfile
+
+from keras.layers import Conv1D, Dense, Flatten, MaxPool1D
+import numpy as np
+import tensorflow as tf
+import tensorframes as tfs
+
+from pyspark.sql.types import Row
+
+from sparkdl.graph.builder import IsolatedSession
+import sparkdl.graph.utils as tfx
+from sparkdl.transformers.tf_tensor import TFTransformer, TFInputGraphBuilder
+
+from ..tests import SparkDLTestCase
+
+def grab_df_arr(df, output_col):
+    """ Stack the numpy array from a DataFrame column """
+    return np.array([row.asDict()[output_col]
+                     for row in df.select(output_col).collect()])
+
+class TFTransformerTest(SparkDLTestCase):
+
+    def _get_rand_vec_df(self, num_rows, vec_size):
+        return self.session.createDataFrame(
+            Row(idx=idx, vec=np.random.randn(vec_size).tolist())
+            for idx in range(num_rows))
+
+    def test_build_from_tf_graph(self):
+        # Build a simple input DataFrame
+        vec_size = 17
+        num_vecs = 31
+        df = self._get_rand_vec_df(num_vecs, vec_size)
+        analyzed_df = tfs.analyze(df)
+
+        # Build the TensorFlow graph
+        with tf.Session() as sess:
+            #x = tf.placeholder(tf.float64, shape=[None, vec_size])
+            x = tfs.block(analyzed_df, 'vec')
+            z = tf.reduce_mean(x, axis=1)
+            graph = sess.graph
+
+            # Get the reference data
+            _results = []
+            for row in df.collect():
+                arr = np.array(row.vec)[np.newaxis, :]
+                _results.append(sess.run(z, {x: arr}))
+            out_ref = np.hstack(_results)
+
+        # Apply the transform
+        gin_from_graph = TFInputGraphBuilder.fromGraph(graph)
+        for gin in [gin_from_graph, graph]:
+            transfomer = TFTransformer(
+                tfInputGraph=TFInputGraphBuilder.fromGraph(graph),
+                inputMapping={
+                    'vec': x
+                },
+                outputMapping={
+                    z: 'outCol'
+                })
+            final_df = transfomer.transform(analyzed_df)
+            out_tgt = grab_df_arr(final_df, 'outCol')
+            self.assertTrue(np.allclose(out_ref, out_tgt))
+
+
+    def test_build_from_saved_model(self):
+        # Setup dataset
+        vec_size = 17
+        num_vecs = 31
+        df = self._get_rand_vec_df(num_vecs, vec_size)
+        analyzed_df = tfs.analyze(df)
+        input_col = 'vec'
+        output_col = 'outputCol'
+
+        # Setup saved model export directory
+        saved_model_root = tempfile.mkdtemp()
+        saved_model_dir = os.path.join(saved_model_root, 'saved_model')
+        serving_tag = "serving_tag"
+        serving_sigdef_key = 'prediction_signature'
+
+        builder = tf.saved_model.builder.SavedModelBuilder(saved_model_dir)
+        with tf.Session(graph=tf.Graph()) as sess:
+            # Model definition: begin
```

Comment:
let's have all the `test_build_...` tests in this class use the same graph. can we refactor so we have a helper function constructing the graph? (i know it's annoying with session etc, but it's possible.)

Reply:
That sounds very useful. Shall we make a separate PR to refactor the `test_build_...` tests?

Reply:
I'd just refactor here. It's not a adding another feature, just making the code more readable and debuggable as it should have been.